content/docs/connect/choose-connection.md

```diff
@@ -0,0 +1,133 @@
+
+- **WebSockets**
+
+  If you require session or interactive transaction support or compatibility with [node-postgres](https://node-postgres.com/) (the popular **npm** `pg` package), use WebSockets. See [Use the driver over WebSockets](/docs/serverless/serverless-driver#use-the-driver-over-websockets).
+
+<Admonition type="note">
+We are working on automatic switching between HTTP and WebSocket as needed. Check our [roadmap](/docs/introduction/roadmap) to see what's coming soon and our Friday [Changelog](/docs/changelog) for the features-of-the-week.
+</Admonition>
+
+### Next, choose your connection type: direct or pooled
+
+You then need to decide whether to use direct connections or pooled connections (using PgBouncer):
+
+- **In general, use pooled connections whenever you can**
+
+  Pooled connections can efficiently manage high numbers of concurrent client connections, up to 10,000. This 10K ceiling works best for serverless applications and application-side connection pools that have many open connections, but infrequent and/or short transactions.
+
+- **Use direct (unpooled) connections if you need persistent connections**
+
+  If your application is focused mainly on tasks like migrations or administrative operations that require stable and long-lived connections, use an unpooled connection.
+
+<Admonition type="note">
+Connection pooling is not a magic bullet. PgBouncer can keep many client connections open (up to 10,000) concurrently, but only a limited number of these can be actively querying the PostgreSQL server at any given time: 64 active backend transactions per user-database pair, as set by the PgBouncer's `default_pool_size` setting. For example, the Postgres user `alex` can hold up to 64 backend connections to a single database at one time.
+</Admonition>
+
+For more information on these choices, see:
+
+- [Neon Serverless Driver](/docs/serverless/serverless-driver)
+- [Connection pooling](/docs/connect/connection-pooling)
+
+## Common Pitfalls
+
+Here are some key points to help you navigate potential issues.
+| Issue | Description |
+|----------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Double pooling | Avoid using client-side pooling if you're using a pooled Neon connection (supported by PgBouncer). Just let Neon handle the pooling to prevent retaining unused connections on the client side. If you must use client-side pooling, make sure connections are released back to the client-side pool early enough to avoid conflicts with PgBouncer. |
```

Comment:
We mentioned above that the 10K limit works well for application-side connection pooling, but here we say to avoid using client-side pooling with a pooled connection.

Reply:
Ok, I'll clarify this. The thing to avoid is pooling that's managed within the client library, doubled up with pooling provided by the application's connections to pgbouncer: if you want to use both (ie double-pooling) you need to configure the client-side pooling more carefully.